sklearn/model_selection/_validation.py

```diff
@@ -247,9 +247,12 @@ def cross_validate(estimator, X, y=None, *, groups=None, scoring=None, cv=None,
             error_score=error_score)
         for train, test in cv.split(X, y, groups))
 
-    results = _aggregate_score_dicts(results)
     if return_estimator:
-        fitted_estimators = results["estimator"]
+        # Extract the estimator objects for each cv split because the
+        # "_aggregate_score_dicts" converts nested estimators to arrays
+        fitted_estimators = [result["estimator"] for result in results]
```

Comment:
yes I think that this would be the job of `_aggregate_score_dicts` to take care. We could make it to only apply `asarray` on time and score info (cf l.1562 of `_validation.py`).

```python    
return {key: np.asarray([score[key] for score in scores])
            for key in scores[0]
            if key in ("time", "score")}
```

Reply:
arff I did not put it in the right place

Reply:
```python
return {
        key: np.asarray([score[key] for score in scores])
        if key.endswith(tuple(["time", "scores"]))
        else [score[key] for score in scores]
        for key in scores[0]
    }
```

Reply:
Do not worry, I know what you mean. I will also add a test for this issue!

Reply:
We can add a non-regression test with a pipeline in `cross_validate` and check that we have a proper list and pipeline as expected.

Reply:
I have applied the suggested changes. Thanks @glemaitre!